lib/Wikia/src/Service/User/Attributes/maintenance/cleanupMigratedAttributes.php

```diff
@@ -42,21 +44,39 @@ private function printAttrsToBeDeleted() {
             $this->output( "NOT DRY RUN, ATTRIBUTES WILL BE DELETED\n" );
         }
         $this->output( "Deletion will begin in 5 seconds...\n" );
-        sleep(5);
+        sleep( 5 );
     }
     
     private function deleteAttributes() {
-        global $wgExternalSharedDB, $wgPublicAttributes;
+        global $wgExternalSharedDB, $wgPublicUserAttributes;
+
+        $db = wfGetDB( DB_MASTER, [], $wgExternalSharedDB );
+        foreach ( $wgPublicUserAttributes as $attribute ) {
+            $this->deleteAttribute( $attribute, $db );
+        }
+    }
+
+    private function deleteAttribute( $attributeName, DatabaseMysqli $db ) {
+        $this->output( "Deleting $attributeName\n" );
 
         if ( $this->dryRun ) {
             return;
         }
 
-        $db = wfGetDB( DB_MASTER, [], $wgExternalSharedDB );
-        ( new WikiaSQL() )
-            ->DELETE( 'user_properties' )
-            ->WHERE( 'up_property' )->IN( $wgPublicAttributes )
-            ->run( $db );
+        $rowsRemaining = true;
+        while ( $rowsRemaining ) {
```

Comment:
Good point. If we were trying to ensure this was being run in a transaction, then yes, I think it should be. In this case though, I was more concerned with transactions being ended after each batch delete. My thinking was that if autocommit was off, then there'd be a chance that these batches wouldn't be commited and then we'd lose the whole advantage of doing it iteratively.

I just double checked though, and it looks like autocommit is on, so we can remove that `commit()` altogether:
```
mysql@master.db-sharedb.service.consul[wikicities]>select @@autocommit;
+--------------+
| @@autocommit |
+--------------+
|            1 |
+--------------+
1 row in set (0.00 sec)
```